src/client-scopes/details/SearchFilter.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  Select,
+  SelectOption,
+  ToolbarItem,
+} from "@patternfly/react-core";
+import { FilterIcon } from "@patternfly/react-icons";
+
+import {
+  AllClientScopes,
+  AllClientScopeType,
+  clientScopeTypesSelectOptions,
+} from "../../components/client-scope/ClientScopeTypes";
+import type { Row } from "../../clients/scopes/ClientScopes";
+
+export type SearchType = "name" | "type" | "protocol";
+export const Protocols = ["all", "saml", "openid-connect"];
```

Comment:
Since this is variable it should start with a lower case.

```suggestion
export const protocols = ["all", "saml", "openid-connect"];
```

Or if you want to use a constant case:
```suggestion
export const PROTOCOLS = ["all", "saml", "openid-connect"];
```

Reply:
Note that you can also tell the TypeScript compiler that the type is a readonly tuple. So instead of having a type of `string[]` it will become `readonly ["all", "saml", "openid-connect"]`.

This also will reflects down to `ProtocolType` which will be of type `"all" | "saml" | "openid-connect"` instead of type `string`.

```suggestion
export const protocols = ["all", "saml", "openid-connect"] as const;
```